janus_core/helpers/utils.py

```diff
@@ -1,11 +1,154 @@
 """Utility functions for janus_core."""
 
+from abc import ABC
 from pathlib import Path
 from typing import Optional
 
 from ase import Atoms
 from spglib import get_spacegroup
 
+from janus_core.helpers.janus_types import PathLike
+
+
+class FileNameMixin(ABC):  # pylint: disable=too-few-public-methods
+    """
+    Provide mixin functions for standard filename handling.
+
+    Parameters
+    ----------
+    struct : Atoms
+        Structure from which to derive the default name if struct_name not provided.
+    struct_name : Optional[str]
+        Struct name to use.
+    file_prefix : Optional[PathLike]
+        Default prefix to use.
+    *additional
+        Components to add to file_prefix (joined by hyphens).
+
+    Methods
+    -------
+    _get_default_struct_name(struct, struct_name)
+         Return the name from the provided struct_name or generate from struct.
+    _get_default_prefix(file_prefix, struct_name)
+         Return a prefix from the provided file_prefix or from struct_name.
+    _build_filename(suffix, *additional, filename, prefix_override)
+         Return a standard format filename if filename not provided.
+    """
+
+    def __init__(
+        self,
+        struct: Atoms,
+        struct_name: Optional[str],
+        file_prefix: Optional[PathLike],
+        *additional,
+    ):
+        """
+        Provide mixin functions for standard filename handling.
+
+        Parameters
+        ----------
+        struct : Atoms
+            Structure from which to derive the default name if struct_name not provided.
+        struct_name : Optional[str]
+            Struct name to use.
+        file_prefix : Optional[PathLike]
+            Default prefix to use.
+        *additional
+            Components to add to file_prefix (joined by hyphens).
+
+        Methods
+        -------
+        _get_default_struct_name(struct, struct_name)
+             Return the name from the provided struct_name or generate from struct.
+        _get_default_prefix(file_prefix, struct_name)
+             Return a prefix from the provided file_prefix or from struct_name.
+        _build_filename(suffix, *additional, filename, prefix_override)
+             Return a standard format filename if filename not provided.
+        """
+        self.struct_name = self._get_default_struct_name(struct, struct_name)
+
+        self.file_prefix = "-".join(
+            (self._get_default_prefix(file_prefix, self.struct_name, *additional),)
```

Comment:
What is this doing? From quick tests it doesn't seem to modify what's returned by `_get_default_prefix`?